libbeat/outputs/elasticsearch/elasticsearch.go

```diff
@@ -50,19 +52,38 @@ var (
 )
 
 type callbacksRegistry struct {
-	callbacks []connectCallback
+	callbacks map[uuid.UUID]connectCallback
```

Comment:
This changes order in which callbacks will be called. I don't think we don't have any ordering requirements yet, but it's something we need to be aware of in the future: Callbacks must not depend on results of a former callback.

Reply:
I will add a comment, so it's less likely we forget it.